### ISSUE TITLE
Refactor orch_expr

### ISSUE THREAD

**Issue by TengHu** [TengHu/ActionWeaver#67]
Need to make changes in chat.py in azure, openai/functions, openai/tools

- Need to have a single place of orchestration definition.
- Support action force invoke.
- Think about how to support sequential function calling within the function dispatch loop.
- In cases where there are no specified actions within an orchestration, the flow will default to the expression defined in orchestration[default_expr], enable calling the same action repeatedly.
- Basically this is where user can control what actions come next after each action invoked.
